spring-cloud-modules/spring-cloud-openfeign/src/main/java/com/baeldung/cloud/openfeign/customizederrorhandling/exception/ProductExceptionHandler.java

```diff
@@ -0,0 +1,40 @@
+package com.baeldung.cloud.openfeign.customizederrorhandling.exception;
+
+import com.baeldung.cloud.openfeign.exception.BadRequestException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@RestControllerAdvice
+public class ProductExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler({ProductServiceNotAvailableException.class})
+    public ResponseEntity<ErrorResponse> handleProductServiceNotAvailableException(ProductServiceNotAvailableException exception, WebRequest request) {
+        return new ResponseEntity<>(new ErrorResponse(
+                HttpStatus.INTERNAL_SERVER_ERROR,
```

Comment:
The continuation indent should be two spaces

Reply:
changed to two spaces